Sources/UI/NotificationCenter/NotificationCenterViewController.swift

```diff
@@ -424,3 +346,152 @@ extension NotificationCenterViewController: UIViewControllerTransitioningDelegat
+        } catch {
+            os_log("Problem fetching notifications list: %s", log: .ui, error.localizedDescription)
+        }
+    }
+    
+    /**
+     Returns a filtered list of notifications from dataStore.notifications. The default implementation filters out notifications that aren't notification center enabled, are deleted or have expired. This method is called automatically by the NotificationCenterTableView and the results are cached for performance.
+     
+     You can override this method if you wish to modify the rules used to filter notifications. For example if you wish to include expired notifications in the table view and instead show their expired status with a visual indicator.
+     */
+    open var notificationFilterPredicate: NSPredicate {
+        return NSPredicate()
+    }
+    
+    func notificationAt(indexPath: IndexPath) -> RoverData.Notification {
+        return fetchedResultsController.object(at: indexPath) as RoverData.Notification
+    }
+    
+    open func cellReuseIdentifier(at indexPath: IndexPath) -> String {
+        return "notification"
+    }
+    
+    // MARK: UITableViewDataSource
+    
+    open func numberOfSections(in tableView: UITableView) -> Int {
+        return self.fetchedResultsController.sections!.count
+    }
+    
+    open func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let sections = self.fetchedResultsController.sections else {
+            assertionFailure("No sections in fetchedResultsController.")
+            return 0
+        }
+        let sectionInfo = sections[section]
+        return sectionInfo.numberOfObjects
+    }
+    
+    open func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let reuseIdentifier = self.cellReuseIdentifier(at: indexPath)
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        
+        guard let notificationCell = cell as? NotificationCell else {
+            return cell
+        }
+        
+        let notification = self.notificationAt(indexPath: indexPath)
+        notificationCell.configure(with: notification, imageStore: self.imageStore)
+        return notificationCell
+    }
+    
+    open func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        guard let sectionInfo = fetchedResultsController.sections?[section] else {
+            return nil
+        }
+        return sectionInfo.name
+    }
+    
+    open func sectionIndexTitles(for tableView: UITableView) -> [String]? {
+        return fetchedResultsController.sectionIndexTitles
+    }
+    
+    open func tableView(_ tableView: UITableView, sectionForSectionIndexTitle title: String, at index: Int) -> Int {
+        return fetchedResultsController.section(forSectionIndexTitle: title, at: index)
+    }
+}
+
+extension NotificationsDataSource : NSFetchedResultsControllerDelegate {
```

Comment:
These methods all operate on the `subscribedTableView` property which from what I can tell is never set. If you take my suggestion of moving the `NotificationsDataSource` functionality to the `NotificationCenterViewController` itself this problem goes away.